### DISCUSSION TITLE
Question about  src/app.cr require order.

### DISCUSSION THREAD

**Discussion by zw963** [luckyframework/lucky#1699] [Q&A] [answered]
I want to use avram with my kemal project, so, i consult a lucky project. 

When i check src/app.cr, i saw this:

```cr
require "./shards"

# Load the asset manifest
Lucky::AssetHelpers.load_manifest "public/mix-manifest.json"

require "../config/server"
require "../config/**"
require "./app_database"
require "./models/base_model"
# ...
```

My question is, we require `../config/**` first, then require `./app_database` later.
but, we use `AppDatabase` in `../config/datbase.cr`, like this:

```cr
AppDatabase.configure do |settings|
	# ...
  end
end
```

Then, we define/open `AppDatabase` class in `./app_database`

```
class AppDatabase < Avram::Database
end
```

Yes, we use a constant before define/open it, if this is intended?

Why we write code like this?

i found `AppDatabase` was defined in `authentic` shards, but this shards is not included
in avram, so, maybe define/open class first, then, config it, is more better?

Thank you.

**Comment by jwoertink** [comment]
Here's an example of using Avram outside of a Lucky project

https://github.com/jwoertink/crystal_orm_test/blob/master/src/orm_test/avram/setup.cr

**Reply by jwoertink** [reply]
oh, and the setup for it https://github.com/jwoertink/crystal_orm_test/blob/master/src/orm_test/avram.cr

**Comment by zw963** [comment]
Thank you very much, in fact, i done it, i just expect to point out, 

for the generated app.cr,  following is better to understood.

```cr
 require "./shards"
 # ...
require "../config/server"
require "./app_database"
require "../config/**"
require "./models/base_model"
# ...
```

Instead of

```cr
 require "./shards"
 # ...
require "../config/server"
require "../config/**"
require "./app_database"
require "./models/base_model"
# ...
```

**Reply by jwoertink** [reply]
Good catch. We can probably change that :smile:

**Comment by zw963** [comment]
Okay, thanks

Should i create i issue point to here, or, convert this discuss to issue directly?

**Comment by zw963** [comment]
Okay, thanks

Should i create i issue point to here, or, convert this discuss to issue directly?

**Reply by jwoertink** [reply, marked as answer]
I've created an issue: 
https://github.com/luckyframework/lucky_cli/issues/768
